lib/rspec/core/output_wrapper.rb

```diff
@@ -0,0 +1,26 @@
+module RSpec
+  module Core
+    # @private
+    class OutputWrapper
+      # @private
+      attr_writer :output
+
+      # @private
+      def initialize(output)
+        @output = output
+      end
+
+      # Redirect calls for IO interface methods
+      (IO.instance_methods - Enumerable.instance_methods - Object.instance_methods + [:to_str]).each do |method|
+        define_method(method) do |*args, &block|
+          @output.send(method, *args, &block)
+        end
+      end
+
+      # @private
+      def respond_to?(*args)
+        args[0] == :output= || @output.respond_to?(*args)
+      end
```

Comment:
Exists case when `output` will be an string with output file name.
In that case `wrapper` should not respond to `IO` methods and i added `respond_to?` method.
I'm not sure the result implementation is good enough.
May be better to write like this:
```ruby
METHODS = IO.instance_methods - Enumerable.instance_methods - Object.instance_methods

if METHODS.include?(args[0])
  @output.respond_to?(*args)
else
  super
end
```
I guess, the last code looks better, but performance will be little bit worse.